### ISSUE TITLE
RSS leak

### ISSUE THREAD

**Issue by lev-kazakov** [laverdet/isolated-vm#207]
To reproduce, run the following code with `--expose-gc` flag:
```js
const ivm = require('isolated-vm')

const noop = () => {}
const code = '//$0.applyIgnored()' // the leak reproduces even without invoking the function reference

const run = async () => {
    const isolate = new ivm.Isolate()
    const context = await isolate.createContext()
    // await context.evalClosure(code, [<primitive>]) does not leak.
    // there are other things that do leak, such as creating external copies.
    // context.evalClosureIgnored() results in: "Assertion failed: (result.second), function RegisterIsolate, file ../src/isolate/platform_delegate.cc, line 18." after a couple of runs.
    await context.evalClosure(code, [noop], { arguments: { reference: true } })
    context.release()
    isolate.dispose()
}

const bytesToMegabytes = bytes => Math.round(bytes / 1024 / 1024);

(async () => {
    gc()
    const initialRss = process.memoryUsage().rss
    let maxRss = initialRss

    for (let i = 1; i <= 20000 ;++i) {
        await run()
        gc()

        const currentRss = process.memoryUsage().rss
        maxRss = Math.max(maxRss, currentRss)
        console.log(`${i}. rss leak: ${bytesToMegabytes(currentRss - initialRss)} MB, max rss reached: ${bytesToMegabytes(maxRss)} MB`)
    }
})()
```

The output will be ~:
`20000. rss leak: 80 MB, max rss reached: 100 MB`

Tested with isolated-vm 3.3.5, node 12.13.1, 12.18.0, 14.4.0 and 14.10.1.
Heap does not leak.

Thx.

**Comment by lev-kazakov** [comment]
refinement:

leaks:
- `await context.evalClosure('', [1], { arguments: { reference: true } })`
- `await context.evalClosure('', [1], { arguments: { externalCopy: true } })`

does not leak:
- `await context.evalClosure('', [1], { arguments: { copy: true } })`

**Comment by lev-kazakov** [comment]
introduced between `v3.0.0` and `v3.1.0`, i.e `v3.0.0` does not leak.

**Comment by laverdet** [comment]
Thanks for the great report. I fixed this issue and it'll be on npm soon as 3.3.6.
